Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp

```diff
@@ -0,0 +1,157 @@
+                    Bx(i,j,k) = coef1_z * Bx(i,j,k) - coef2_z * Ey(i,j,k);
+                // At the -z boundary (innermost guard cell)
+                if ( k==domain_box.smallEnd(2)-1 )
+                    Bx(i,j,k) = coef1_z * Bx(i,j,k) + coef2_z * Ey(i,j,k+1);
+#elif WARPX_DIM_XZ
+                // At the +z boundary (innermost guard cell)
+                if ( j==domain_box.bigEnd(1)+1 )
+                    Bx(i,j,k) = coef1_z * Bx(i,j,k) - coef2_z * Ey(i,j,k);
+                // At the -z boundary (innermost guard cell)
+                if ( j==domain_box.smallEnd(1)-1 )
+                    Bx(i,j,k) = coef1_z * Bx(i,j,k) + coef2_z * Ey(i,j+1,k);
+#endif
+            }
+        );
+
+        // Apply Boundary condition to By
+        amrex::ParallelFor(tby,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k){
+
+                // At the +x boundary (innermost guard cell)
+                if ( i==domain_box.bigEnd(0)+1 )
+                    By(i,j,k) = coef1_x * By(i,j,k) - coef2_x * Ez(i,j,k);
+                // At the -x boundary (innermost guard cell)
+                if ( i==domain_box.smallEnd(0)-1 )
+                    By(i,j,k) = coef1_x * By(i,j,k) + coef2_x * Ez(i+1,j,k);
+#ifdef WARPX_DIM_3D
+                // At the +z boundary (innermost guard cell)
+                if ( k==domain_box.bigEnd(2)+1 )
+                    By(i,j,k) = coef1_z * By(i,j,k) + coef2_z * Ex(i,j,k);
+                // At the -z boundary (innermost guard cell)
+                if ( k==domain_box.smallEnd(2)-1 )
+                    By(i,j,k) = coef1_z * By(i,j,k) - coef2_z * Ex(i,j,k+1);
+#elif WARPX_DIM_XZ
+                // At the +z boundary (innermost guard cell)
+                if ( j==domain_box.bigEnd(1)+1 )
+                    By(i,j,k) = coef1_z * By(i,j,k) + coef2_z * Ex(i,j,k);
+                // At the -z boundary (innermost guard cell)
+                if ( j==domain_box.smallEnd(1)-1 )
+                    By(i,j,k) = coef1_z * By(i,j,k) - coef2_z * Ex(i,j+1,k);
+#endif
+            }
+        );
+
+        // Apply Boundary condition to Bz
+        amrex::ParallelFor(tbz,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k){
+
+                // At the +x boundary (innermost guard cell)
+                if ( i==domain_box.bigEnd(0)+1 )
+                    Bz(i,j,k) = coef1_x * Bz(i,j,k) + coef2_x * Ey(i,j,k);
+                // At the -x boundary (innermost guard cell)
+                if ( i==domain_box.smallEnd(0)-1 )
+                    Bz(i,j,k) = coef1_x * Bz(i,j,k) - coef2_x * Ey(i+1,j,k);
+#ifdef WARPX_DIM_3D
+                // At the +y boundary (innermost guard cell)
+                if ( j==domain_box.bigEnd(1)+1 )
+                    Bz(i,j,k) = coef1_x * Bz(i,j,k) - coef2_y * Ex(i,j,k);
+                // At the -y boundary (innermost guard cell)
+                if ( j==domain_box.smallEnd(1)-1 )
+                    Bz(i,j,k) = coef1_x * Bz(i,j,k) + coef2_y * Ex(i,j+1,k);
```

Comment:
Looks like a 3D bug:
```suggestion
                    Bz(i,j,k) = coef1_y * Bz(i,j,k) - coef2_y * Ex(i,j,k);
                // At the -y boundary (innermost guard cell)
                if ( j==domain_box.smallEnd(1)-1 )
                    Bz(i,j,k) = coef1_y * Bz(i,j,k) + coef2_y * Ex(i,j+1,k);
```

Reply:
Following Murphy's law, this hid in the only direction of propagation that we did not test.